### PR TITLE
make sure AggregationTest creates reduced aggregations.

### LINKED ISSUE #63925: AggregationsTests.testFromXContentWithRandomFields reproducible failure
This test failed on a PR but it seems it has started today.

**Build scan**:

https://gradle-enterprise.elastic.co/s/drlszluma2dvc (different to the report failure)

**Repro line**:

```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.AggregationsTests.testFromXContentWithRandomFields" -Dtests.seed=413BAB3F569B03D6 -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=Asia/Kabul -Druntime.java=11
```

**Reproduces locally?**:

yes

**Applicable branches**:

Only tried in master

**Failure history**:

It seems this test start failing today

**Failure excerpt**:

```
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.AggregationsTests.testFromXContentWithRandomFields" -Dtests.seed=413BAB3F569B03D6 -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=Asia/Kabul -Druntime.java=11
  2> java.lang.IllegalStateException: aggregation was not reduced
        at __randomizedtesting.SeedInfo.seed([413BAB3F569B03D6:432E867E0D5C8B03]:0)
        at org.elasticsearch.search.aggregations.metrics.InternalScriptedMetric.aggregation(InternalScriptedMetric.java:91)
        at org.elasticsearch.search.aggregations.metrics.InternalScriptedMetric.doXContentBody(InternalScriptedMetric.java:151)
        at org.elasticsearch.search.aggregations.InternalAggregation.toXContent(InternalAggregation.java:299)
        at org.elasticsearch.search.aggregations.Aggregations.toXContentInternal(Aggregations.java:130)
        at org.elasticsearch.search.aggregations.bucket.InternalSingleBucketAggregation.doXContentBody(InternalSingleBucketAggregation.java:160)
        at org.elasticsearch.search.aggregations.InternalAggregation.toXContent(InternalAggregation.java:299)
        at org.elasticsearch.search.aggregations.Aggregations.toXContentInternal(Aggregations.java:130)
        at org.elasticsearch.search.aggregations.bucket.terms.InternalSignificantTerms$Bucket.toXContent(InternalSignificantTerms.java:157)
        at org.elasticsearch.search.aggregations.bucket.terms.InternalMappedSignificantTerms.doXContentBody(InternalMappedSignificantTerms.java:139)
        at org.elasticsearch.search.aggregations.InternalAggregation.toXContent(InternalAggregation.java:299)
        at org.elasticsearch.search.aggregations.Aggregations.toXContentInternal(Aggregations.java:130)

```

### PR DESCRIPTION
**Pull request by iverase**
The test calls ` InternalAggregationTestCase#createTestInstanceForXContent()`  instead  of `InternalAggregationTestCase#createTestInstance()`.  The method needs to become public. 

Fixes #63925

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalScriptedMetricTests.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ protected InternalScriptedMetric createTestInstance(String name, Map<String, Obj
     }
 
     private List<Object> randomAggregations() {
-        return randomList(randomBoolean() ? 1 : 5, this::randomAggregation);
+        return randomList(1, randomBoolean() ? 1 : 5, this::randomAggregation);
```
Comment (jimczi) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalScriptedMetricTests.java:94]: I don't think this is the right fix. The tests that fail are the one that always expect a reduced aggregation so they require a single element. That's why `InternalAggregationTestCase#createTestInstanceForXContent` was added but it's not used in `AggregationsTests`. See https://github.com/elastic/elasticsearch/blob/f20699ead53b8912918f4bda6cbfe89a61b8b983/server/src/test/java/org/elasticsearch/search/aggregations/AggregationsTests.java#L299 where replacing the call with `createTestInstanceForXContent` fixes all the failures.

**Review by jimczi — CHANGES_REQUESTED** [review]

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalScriptedMetricTests.java (hunk 1 above)
Comment (nik9000) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalScriptedMetricTests.java:94]: Yeah. I *think* the correct fix is in AggregationsTests.java to replace:
```
            aggs.add(testCase.createTestInstance());
```
with:
```
            aggs.add(testCase.createTestInstanceForXContent());
```

I had long forgotten about this test!

Comment (iverase) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalScriptedMetricTests.java:94]: ups, thanks @jimczi you are right. I guess we will need to make that method accessible to the test as currently is protected.

**Review by nik9000 — APPROVED** [review]
LGTM

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/1
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Pull request merged**